doc/man7/EVP_KDF-KB.pod

```diff
@@ -0,0 +1,111 @@
+=pod
+
+=head1 NAME
+
+EVP_KDF-KB - The Key-Based EVP_KDF implementation
+
+=head1 DESCRIPTION
+
+The EVP_KDF-KB algorithm implements the Key-Based key derivation function
+(KBKDF).  KBKDF derives a key from repeated application of a keyed MAC to an
+input secret (and other optional values).
+
+=head2 Identity
+
+"KBKDF" is the name for this implementation; it can be used with the
+EVP_KDF_fetch() function.
+
+=head2 Supported parameters
+
+The supported parameters are:
+
+=over 4
+
+=item B<OSSL_KDF_PARAM_PROPERTIES> ("properties") <UTF8 string>
+
+=item B<OSSL_KDF_PARAM_DIGEST> ("digest") <UTF8 string>
+
+=item B<OSSL_KDF_PARAM_KEY> ("key") <octet string>
+
+=item B<OSSL_KDF_PARAM_SALT> ("salt") <octet string>
+
+=item B<OSSL_KDF_PARAM_INFO> ("info") <octet string>
+
+=back
+
+The parameters key, salt, and info correspond to KI, Label, and Context
+(respectively) in SP800-108.  As in that document, salt and info are optional
+and may be omitted.
+
+=head1 NOTES
+
+A context for KBKDF can be obtained by calling:
+
+ EVP_KDF *kdf = EVP_KDF_fetch(NULL, "KBKDF", NULL);
+ EVP_KDF_CTX *kctx = EVP_KDF_CTX_new(kdf);
+
+The output length of an KBKDF is specified via the C<keylen>
+parameter to the L<EVP_KDF-derive(3)> function.
```

Comment:
@levitte should this be `EVP_KDF(3)-derive`?
I'm not sure I yet understand the differences in naming completely.

Reply:
Uhmmm, no, this should be `L<EVP_KDF_derive(3)>`

As for what you're asking...

`EVP_KDF_derive` is an existing function, yeah?  So to get the information, you would do `man EVP_KDF_derive`.

However, `EVP_KDF-KB` indicates that this isn't about an actual symbol (as `EVP_KDF_KB` might make you think), but something else.  The neat thing with dashes for this is that `man` will attempt to find a man-page for multiple arguments by concatenating them with dashes in between, so `man EVP_KDF KB` is equivalent to `man EVP_KDF-KB` and becomes a neat way to express that this is an `EVP_KDF` named "KB".

Reply:
My FreeBSD man(1) does not have this behavior; multiple arguments are treated as independent manuals to display in sequence (after close of pager)

Reply:
It's quite possible the concatenation is a Linux thing.  However, nothing stops you from typing `man EVP_KDF-SSHKDF`, it works as well on all platforms, as expected.

However, the fact remains that man3 manuals should be named the same way as the functions it describes.  Since the function in question is called `EVP_KDF_derive`, the manual should be `EVP_KDF_derive(3)`.  That is, unless we can suddenly use dashes in C symbols :wink:

(incidently, perl6 supports dashes and apostrophes in symbol names...  that'll make for interesting page references :laughing:)